### PR TITLE
fix(cli): new from external module with multiple types

### LINKED ISSUE #887: PROJECT-TYPE cli argument not working for external modules with multiple types
`ProjectType` is being used for two different concepts:

1. As a definition for a projen project type: https://github.com/projen/projen/blob/main/src/inventory.ts#L26
2. As an enum used for jsii purposes: https://github.com/projen/projen/blob/main/src/project.ts#L523

Because all projects inherit from the `Project` class, the `ProjectOptions` includes the `projectType` property: https://github.com/projen/projen/blob/main/src/project.ts#L84

The `new` cli command exposes `PROJECT-TYPE` as an argument, which gets camelcase-d to `projectType`: https://github.com/projen/projen/blob/main/src/cli/cmds/new.ts#L18

This argument is intended to be used to determine the projen project class to use. But it also sets the `ProjectOption.projectType` property. The config for this property is:

```
{
    "path": ["projectType"],
    "parent": "ProjectOptions",
    "name": "projectType",
    "fqn": "projen.ProjectType",
    "docs": "Which type of project this is (library/app).",
    "type": "ProjectType",
    "kind": "enum",
    "switch": "project-type",
    "default": "ProjectType.UNKNOWN",
    "optional": true,
    "featured": true
}
```

The above clashes with the idea of `projectType` representing the [projen project type](https://github.com/projen/projen/blob/main/src/inventory.ts#L26).

As a result, I am seeing the following error when bootstrapping a new projen project using an external module with multiple project types defined:

```
npx projen new --from "file:../my-projen-types" foo
```

results in

```
evalmachine.<anonymous>:6
  projectType: ProjectType.FOO,
               ^

ReferenceError: ProjectType is not defined
    at evalmachine.<anonymous>:6:16
```

External modules work for a single type, because `PROJECT-TYPE` is not required and thus does not set the `projectType` option, which then results in the default (`ProjectType.UNKNOWN`).

### PR DESCRIPTION
**Pull request by Chriscbr**
Fixes #887

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Chriscbr** [comment]
To reproduce the original bug, I tried using @taimos/projen (thanks @hoegertn) -- though doing it the naive way fails for me:

```
npx projen new --from @taimos/projen taimos-ts-lib
```

with the error: `Error: cannot find module 'projen'`.

If I try `npm i projen && npx projen new ...` instead, I get the error: `Error: No projects found after installing @taimos/projen. The module must export at least one class which extends projen.Project`.

I'm still not sure what causes this. But once that's fixed, I think it would be good to use this as an integration test of the `npx projen new --from <path> <pjid>` flow.

But I was able to work around this by downloading @taimos/projen locally, building it to generate the .jsii, and running:

```
npx projen new --from "file:../taimos-projen" taimos-ts-lib
```

This yields the error message from #887, so I was able to test that this PR fixes the bug.

**Comment by eladb** [comment]
> with the error: Error: cannot find module 'projen'.

I think this is a bug we have at the moment with bootstrapping projects from external modules. It's a result of the fact that the CLI does not add a `projen` symlink to `node_modules` now. I think the solution is to also add `projen` when bootstrapping from an external module.

File: src/cli/cmds/new.ts [hunk 1]
```diff
@@ -11,11 +11,11 @@ import { exec } from '../../util';
 import { tryProcessMacro } from '../macros';
 
 class Command implements yargs.CommandModule {
-  public readonly command = 'new [PROJECT-TYPE] [OPTIONS]';
+  public readonly command = 'new [PROJECT-TYPE-NAME] [OPTIONS]';
```
Comment (Chriscbr) [inline src/cli/cmds/new.ts:14]: `PROJECT-TYPE-ID` or `PROJEN-ID` or some other variation could probably work just as well (just thought of this now...)

**Comment by Chriscbr** [comment]
@eladb thanks, that fixed the issue. Added an integ test.

**Comment by gradybarrett** [comment]
Installing projen should fix most of #648. Not sure if the peer dep discrepancies have been solved by any recent PRs, but if so then you could close #648.

**Comment by Chriscbr** [comment]
From #648:

> When people consume that external project
> Then they should be able to bring their desired projen version, and it should not clash with the version of projen used to build the external project.

This PR partially satisfies that - though we might have to figure out what counts as bringing a desired projen version. With these changes, if you run `yarn add --dev projen@x.y.z` before `npx projen new --from <module> <pjid>`, it will respect the version of projen you specified.

But it will not automatically pick up if you have a specific version of projen installed globally via `yarn global add projen`.

**Comment by gradybarrett** [comment]
I'm not sure we have to satisfy that at bootstrap-time though. I think it's sufficient to use latest projen to bootstrap, at which point folks can choose the version they want and regenerate.

The only caveat is if someone passes a projen version in as a CLI option, then we could use that version during bootstrapping couldn't we?

**Comment by Chriscbr** [comment]
Cool, I think that makes sense. It should be possible to use the verison doing bootstrapping as well, I'll take a look.

**Review by eladb — APPROVED** [review]

**Pull request merged**